src/runtime-rs/crates/runtimes/virt_container/src/container_manager/io/shim_io.rs

```diff
@@ -61,7 +73,11 @@ impl ShimIo {
                 .open(&stdin)
                 .await
             {
-                Ok(file) => Some(Box::new(file)),
+                Ok(file) => {
+                    // Set it to blocking to avoid infinitely handling EAGAIN when the reader is empty
+                    set_flag_with_blocking(file.as_raw_fd());
```

Comment:
Why not just remove the `O_NONBLOCK` flag from the above open options?

Reply:
Since the stdin peer point (which is hold by containerd) could not be openned immediately, which would block here's open with block mode, and we wouldn't want to block here, thus here opened with nonblock and then reset it to block mode for tokio async io. 

@quanweiZhou  You'd better comment it clearly why opened it with  "O_NONBLOCK" mode.

Reply:
Thanks, @bergwolf @lifupan add comment for this.

Reply:
Good point!